.github/workflows/deploy.yml

```diff
@@ -0,0 +1,45 @@
+name: Deploy to Netlify
+
+on:
+  push:
+    branches: ['main']
+  workflow_dispatch:
+    inputs:
+      deploy-msg:
+        description: 'Deploy to Production'
+        required: true
+
+jobs:
+  deploy:
+    name: Build and Deploy to Netlify
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.10.0
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout repository
+        uses: actions/checkout@v3.2.0
+
+      - name: Install modules
+        run: npm ci
+
+      - name: Decode .env file
+        run: echo "${{ secrets.ENV_FILE_CONTENT }}" | base64 -d > .env
```

Comment:
@tyrro I have replied a similar comment [here](https://github.com/manh-t/react-survey/pull/41#discussion_r1223949092).

For more details, I follow a practice from mobile side to hide the `.env` file which contains the sensitive configuration like `client secret`, `client id` from the codebase. Because we do not often push `.env` file to the version control, right? So there should be a way to get the `.env` file to build.

Since we don't have any server to store it so I decide to store this file in the secret environment of GA instead 🙇 

I'm a new guy here so if this approach is not optimized, feel free to suggest other ways 😁 cc @nvminhtue

Reply:
I reply in this thread to keep you all can follow. I think you don't need to upload the `env` file as well but use the target platform (`netlify` in your case) to control the ENVs instead.
Besides, how could you encode when you already ignored the `.env` file?